app/container/aws-ecs-ec2/provider.go

```diff
@@ -0,0 +1,223 @@
+package aws_ecs_ec2
+
+import (
+	"context"
+	"fmt"
+	"gopkg.in/nullstone-io/go-api-client.v0"
+	"gopkg.in/nullstone-io/go-api-client.v0/types"
+	"gopkg.in/nullstone-io/nullstone.v0/app"
+	"gopkg.in/nullstone-io/nullstone.v0/app/container/aws-ecr"
+	"gopkg.in/nullstone-io/nullstone.v0/config"
+	"gopkg.in/nullstone-io/nullstone.v0/outputs"
+	"log"
+	"os"
+)
+
+var (
+	logger = log.New(os.Stderr, "", 0)
+)
+
+var ModuleContractName = types.ModuleContractName{
+	Category:    string(types.CategoryApp),
+	Subcategory: string(types.SubcategoryAppContainer),
+	Provider:    "aws",
+	Platform:    "ecs",
+	Subplatform: "ec2",
+}
+
+var _ app.Provider = Provider{}
+
+type Provider struct {
+}
+
+func (p Provider) DefaultLogProvider() string {
+	return "cloudwatch"
+}
+
+func (p Provider) identify(nsConfig api.Config, details app.Details) (*InfraConfig, error) {
+	logger.Printf("Identifying infrastructure for app %q\n", details.App.Name)
+	ic := &InfraConfig{}
+	retriever := outputs.Retriever{NsConfig: nsConfig}
+	if err := retriever.Retrieve(details.Workspace, &ic.Outputs); err != nil {
+		return nil, fmt.Errorf("Unable to identify app infrastructure: %w", err)
+	}
+	ic.Print(logger)
+	return ic, nil
+}
+
+func (p Provider) Push(nsConfig api.Config, details app.Details, userConfig map[string]string) error {
+	return (aws_ecr.Provider{}).Push(nsConfig, details, userConfig)
+}
+
+// Deploy takes the following steps to deploy an AWS ECS service
+//   Get task definition
+//   Change image tag in task definition
+//   Register new task definition
+//   Deregister old task definition
+//   Update ECS Service (This always causes deployment)
+func (p Provider) Deploy(nsConfig api.Config, details app.Details, userConfig map[string]string) error {
+	ic, err := p.identify(nsConfig, details)
+	if err != nil {
+		return err
+	}
+
+	logger.Printf("Deploying app %q\n", details.App.Name)
+	version := userConfig["version"]
+	if version == "" {
+		return fmt.Errorf("no version specified, version is required to deploy")
+	}
+
+	logger.Printf("Updating app version to %q\n", version)
+	if err := app.CreateDeploy(nsConfig, details.App.StackId, details.App.Id, details.Env.Id, version); err != nil {
```

Comment:
We will have to revisit this as we rebase into our current work for deployments. This call to create a deploy in the db gets removed from here.